Sources/MessagingPush/CustomerIOParsedPushPayload.swift

```diff
@@ -29,19 +29,27 @@ public class CustomerIOParsedPushPayload {
 
     public var deepLink: URL? {
         get {
-            cio.push.link?.url
+            cio?.push.link?.url
         }
         set {
-            cioPush = cioPush.linkSet(newValue?.absoluteString)
+            if cioPush == nil {
+                cio = CioRichPushPayload(push: CioRichPushPayload.Push(link: newValue?.absoluteString, image: nil))
+            } else {
+                cioPush = cioPush?.linkSet(newValue?.absoluteString)
```

Comment:
```
private func updateCioPush(link: String?, image: String?) {
    if let cioPush = cioPush {
        if let link = link {
            self.cioPush = cioPush.linkSet(link)
        }
        if let image = image {
            self.cioPush = cioPush.imageSet(image)
        }
    } else {
        cio = CioRichPushPayload(push: CioRichPushPayload.Push(link: link, image: image))
    }
}

```

does combining logic from both methods simplify this? i find this version easier to understand but its totally subjective.

Reply:
I like the suggestion, easier to understand for me as well

Reply:
Thanks for the suggestion. 

I was able to reduce the complexity of the getter/setter to not need the original logic or suggested logic.